cleanlab_studio/cli/api_service.py

```diff
@@ -19,6 +20,9 @@
 base_url = os.environ.get("CLEANLAB_API_BASE_URL", "https://api.cleanlab.ai/api/cli/v0")
 
 
+MAX_PARALLEL_UPLOADS = 100  # XXX choose this dynamically?
```

Comment:
what would the optimal value for this depend on? are they things we can know before starting the upload?

I could see it making sense to do something like `min(<num_files> / <some_factor>, <max_allowed_parallel_uploads>)`